apps/namadillo/src/atoms/balance/functions.ts

```diff
@@ -56,3 +59,57 @@ export const getTotalDollar = (list?: TokenBalance[]): BigNumber | undefined =>
 export const getTotalNam = (list?: TokenBalance[]): BigNumber =>
   list?.find((i) => i.asset.base === namadaAsset.base)?.balance ??
   new BigNumber(0);
+
+const tnamAddressToDenomTrace = (
+  address: string,
+  tokenAddresses: (NativeToken | IbcToken)[]
+): DenomTrace | undefined => {
+  const token = tokenAddresses.find((entry) => entry.address === address);
+  const trace = token && "trace" in token ? token.trace : undefined;
+
+  if (typeof trace === "undefined") {
+    return undefined;
```

Comment:
For this case, we can return the `nam` denom (or `namnam`), because it'll be the `NativeToken`. 

@mateuszjasiuk just in case, can you confirm this again?
On the indexer api, the endpoint `/token`, if a token doesn't have a `trace` value it means that it's the native token, meaning the `NAM` token, correct?

Reply:
In this case, it should return specifically only for IBC tokens because it's used as part of the IBC asset lookup. I'll add a comment to make this clearer.

Reply:
It should mean native token. I will try to figure out why on housefire we have multiple tokens without an ibc trace. If you want to be double sure you can always get native token address from parameters and compare :)